solr/core/src/java/org/apache/solr/cluster/events/impl/ClusterEventProducerFactory.java

```diff
@@ -149,21 +153,25 @@ public void deleted(ContainerPluginsRegistry.ApiInfo plugin) {
           ClusterEventListener listener = (ClusterEventListener) instance;
           clusterEventProducer.unregisterListener(listener);
         } else if (instance instanceof ClusterEventProducer) {
-          // replace the existing impl with NoOp
-          if (cc.getClusterEventProducer() instanceof DelegatingClusterEventProducer) {
-            ((DelegatingClusterEventProducer) cc.getClusterEventProducer())
-                .setDelegate(new NoOpProducer(cc));
+          if (ClusterEventProducer.PLUGIN_NAME.equals(plugin.getInfo().name)) {
```

Comment:
There's quite some duplication between this listener method (`deleted`) and method `added` above. Likely could use some factoring out (and maybe slight rework of the logs to differentiate adding and deleting?).

Reply:
I debated whether to do it in this PR or separately... strictly speaking it's not a part of this PR, but I refactored a similar pattern in a listener that belongs to this PR ... so maybe I'll do it here :)